gnomad_qc/v4/create_release/create_de_novo_release.py

```diff
@@ -42,343 +31,248 @@
+    ht = filter_low_conf_regions(
+        ht, filter_lcr=True, filter_decoy=False, filter_segdup=True
+    )
+    logger.info(
+        f"Getting {ht.count()} de novos after removing low confidence regions..."
     )
 
-    return selects
-
+    diploid_expr, hemi_x_expr, hemi_y_expr = get_copy_state_by_sex(
+        ht.locus,
+        ht.is_xx,
+    )
 
-def custom_an_select_globals(ht: hl.Table) -> Dict[str, hl.expr.Expression]:
-    """
-    Select AN Table globals for release.
+    adj_trio = (
+        hl.case()
+        .when(
+            diploid_expr,
+            ht.proband_entry.adj & ht.father_entry.adj & ht.mother_entry.adj,
+        )
+        .when(hemi_x_expr, ht.proband_entry.adj & ht.mother_entry.adj)
+        .when(hemi_y_expr, ht.proband_entry.adj & ht.father_entry.adj)
+        .or_missing()
+    )
 
-    :param ht: Hail Table with AN.
-    :return: Select expression dict.
-    """
-    return {
-        "exomes_an_globals": hl.struct(
-            meta=ht.strata_meta,
-            sample_count=ht.strata_sample_count,
+    ht = ht.annotate(adj_trio=adj_trio)
+
+    ht = (
+        ht.group_by("locus", "alleles")
+        .aggregate(
+            de_novo_AC=hl.struct(
+                AC_all=hl.agg.count_where(ht.adj_trio),
+                AC_all_raw=hl.agg.count(),
+                AC_high_conf=hl.agg.count_where(
+                    (ht.de_novo_call_info.confidence == "HIGH") & ht.adj_trio
+                ),
+                AC_medium_conf=hl.agg.count_where(
+                    (ht.de_novo_call_info.confidence == "MEDIUM") & ht.adj_trio
+                ),
+                AC_medium_conf_P_0_9=hl.agg.count_where(
+                    (ht.de_novo_call_info.confidence == "MEDIUM")
+                    & (ht.de_novo_call_info.p_de_novo >= 0.9)
+                    & ht.adj_trio
+                ),
+                AC_low_conf=hl.agg.count_where(
+                    (ht.de_novo_call_info.confidence == "LOW") & ht.adj_trio
+                ),
+            ),
+            p_de_novo_stats=hl.agg.stats(ht.de_novo_call_info.p_de_novo),
+            mixed_site=hl.agg.collect(ht.mixed_site)[0],
+            # the mixed_site info should stay the same for each variant.
         )
-    }
+        .key_by("locus", "alleles")
+    )
+    ht = ht.annotate(p_de_novo_stats=ht.p_de_novo_stats.drop("n", "sum")).checkpoint(
+        new_temp_file("denovo_ac", "ht")
+    )
+    logger.info(f"Getting {ht.count()} unique de novo variants after computing AC...")
+
+    ht = ht.annotate(vep=vep_ht[ht.key].vep)
+    ht = process_consequences(ht, has_polyphen=False)
+
+    ht = ht.annotate(
+        pass_filters=filters_ht[ht.key]
+        .filters.intersection(hl.set(["AS_VQSR", "InbreedingCoeff"]))
+        .length()
+        == 0,
+        # AC0 is not used as a criteria in de novos because AC0 was used for
+        # release sites and some of the probands weren't included in the release.
```

Comment:
```suggestion
        pass_filters=filters_ht[ht.key]
        # AC0 is not used as a criteria in de novos because AC0 was used for
        # release sites and some of the probands weren't included in the release.
        .filters.intersection(hl.set(["AS_VQSR", "InbreedingCoeff"]))
        .length()
        == 0,
```
this comment is slightly more intuitive above the code it applies to